client/components/date-range/trigger.js

```diff
@@ -0,0 +1,53 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Gridicon from 'gridicons';
+
+class DateRangeTrigger extends Component {
+	static propTypes = {
+		startDateText: PropTypes.string.isRequired,
+		endDateText: PropTypes.string.isRequired,
+		buttonRef: PropTypes.oneOfType( [
+			PropTypes.func,
+			PropTypes.shape( { current: PropTypes.instanceOf( Button ) } ),
+		] ).isRequired,
+		onTriggerClick: PropTypes.func,
+	};
+
+	static defaultProps = {
+		onTriggerClick: noop,
+	};
+
+	render() {
+		const props = this.props;
+
+		return (
+			<Button
+				className="date-range__trigger-btn"
+				ref={ props.buttonRef }
+				onClick={ props.onTriggerClick }
+				compact
+			>
+				<Gridicon className="date-range__trigger-btn-icon" icon="calendar" />
+				<span className="date-range__trigger-btn-text">
+					{ props.startDateText }
+					&nbsp; - &nbsp;
+					{ props.endDateText }
```

Comment:
Is this formatting (start date, dash, then end date) locale agnostic? I'm not sure if some locales would use a different separator or even put them in different order?

If using dash, should use proper `&ndash;`.

Reply:
Hmmm that _is_ a good point. Options

1. Give consumer a way to format the entire string `startDate + endDate`
2. Give consumer a way to choose separator
3. Find a way to make the separator locale aware (if that is a thing)

Reply:
Found this which also suggests making it open to translators

https://stackoverflow.com/questions/10449487/whats-a-i18n-friendly-way-to-show-a-date-range

Reply:
Resolve based on comments here `p1547464508137900-slack-calypso` and existing code [here](https://github.com/Automattic/wp-calypso/blame/master/client/my-sites/stats/stats-date-picker/index.jsx#L81)